wishlists.py

```diff
@@ -27,35 +27,43 @@ def index():
 def add_wishlist():
-	"""
-	The route for adding new wishlists, specified by userID and name of the wishlist. You can check
-	the POST requests using CURL.
-	Example: curl -i -H 'Content-Type: application/json' -X POST -d '{"name":"Xynazog","user_id":123}' http://127.0.0.1:5000/wishlists
-	H is for headers, X is used to specify HTTP Method, d is used to pass a message.
-	"""
-
-	name = request.json['name']
-	uid = request.json['user_id']
-	try:
-		return db.create_wishlist(name,uid), HTTP_201_CREATED
-	except WishlistException:
-		return jsonify(message='Cannot create a new wishlist named %s' % name), HTTP_400_BAD_REQUEST
+    """
+    The route for adding new wishlists, specified by userID and name of the wishlist. You can check
+    the POST requests using CURL.
+    Example: curl -i -H 'Content-Type: application/json' -X POST -d '{"name":"Xynazog","user_id":123}' http://127.0.0.1:5000/wishlists
+    H is for headers, X is used to specify HTTP Method, d is used to pass a message.
+    """
+    data = request.get_json()
+    if is_valid(data,'wishlist'):
+        name = request.json['name']
+        uid = request.json['user_id']
+        try:
+            return db.create_wishlist(name,uid), HTTP_201_CREATED
+        except WishlistException:
+            return jsonify(message='Cannot create a new wishlist named %s' % name), HTTP_400_BAD_REQUEST
+    else:
+        message = {'error' : 'Wishlist data was not valid'}
+        return jsonify(message), HTTP_400_BAD_REQUEST
 
 @app.route('/wishlists/<int:wishlist_id>/items',methods=['POST'])
 def add_item_to_wishlist(wishlist_id):
-	"""
-	The route for adding new items to the wishlist. This method can also be checked using CURL.
-	Pre-requisite: Create a wishlist to add an item.
-	Example: curl -i -H 'Content-Type: application/json' -X POST -d '{"id":"i123","description":"Awesome product!"}' http://127.0.0.1:5000/wishlists/1/items
-	"""
-
-	tempDic = {}
-	tempDic['id'] = request.json['id']
-	tempDic['description'] = request.json['description']
-	try:
-		return db.add_item(wishlist_id,tempDic), HTTP_201_CREATED
-	except WishlistException:
-		return jsonify(message='Cannot add a new item %s' % request.json['id']), HTTP_400_BAD_REQUEST
+    """
+    The route for adding new items to the wishlist. This method can also be checked using CURL.
+    Pre-requisite: Create a wishlist to add an item.
+    Example: curl -i -H 'Content-Type: application/json' -X POST -d '{"id":"i123","description":"Awesome product!"}' http://127.0.0.1:5000/wishlists/1/items
+    """
+    data = request.get_json()
+    if is_valid(data,'item'):
+        try:
+            tempDic = {}
+            tempDic['id'] = request.json['id']
+            tempDic['description'] = request.json['description']
+            return db.add_item(wishlist_id,tempDic), HTTP_201_CREATED
+        except WishlistException:
+            return jsonify(message='Cannot add a new item %s' % request.json['id']), HTTP_400_BAD_REQUEST
```

Comment:
`        except WishlistException:
            message = { 'error' : 'Wishlist %s was not found' % wishlist_id }
            return jsonify(message), HTTP_404_NOT_FOUND`